force-app/main/default/classes/DMLOperations.cls

```diff
@@ -132,8 +153,23 @@ public with sharing class DMLOperations {
             // Set the Opportunity's Stage to Qualification
             // Set the Opportunity's Close Date to 3 months from today
             // Set the Opportunity's Amount to 50000
-
+            
+            /*List<Opportunity> oppsList = new List<Opportunity>();
+              So here I don't have to create a new List.
+              Updating `opp` will update `opportunities` List.
+              I can just `upsert opportunities` after.
+            */
+        
```

Comment:
You are correct; they refer to the same place in memory, and collections are passed in as an array of pointers.